### ISSUE TITLE
[3.3.5] DB/Creature: missing spawn creature 40405 Kieupid <Pet Trainer>

### ISSUE THREAD

**Issue by tkrokli** [TrinityCore/TrinityCore#17747]
**Description**:
Kieupid `<Pet Trainer>`, creature 40405, is missing from the `creature` table.
http://www.wowhead.com/npc=40405/kieupid
http://wow.gamepedia.com/Kieupid
- ![WotLK](https://hydra-media.cursecdn.com/wow.gamepedia.com/0/08/Wrath-Logo-Small.PNG "Wrath of the Lich King") **Patch 3.3.3 (2010-03-23): Added.**

**Current behaviour**: Tell us what happens
From the table `creature_template` :
```sql
SELECT * FROM `creature_template` WHERE `entry` = 40405;
```
```
entry  difficulty_entry_1  difficulty_entry_2  difficulty_entry_3  KillCredit1  KillCredit2  modelid1  modelid2  modelid3  modelid4  name     subname      IconName  gossip_menu_id  minlevel  maxlevel     exp  faction  npcflag  speed_walk  speed_run   scale    rank  dmgschool  BaseAttackTime  RangeAttackTime  BaseVariance  RangeVariance  unit_class  unit_flags  unit_flags2  dynamicflags  family  trainer_type  trainer_spell  trainer_class  trainer_race    type  type_flags  lootid  pickpocketloot  skinloot  resistance1  resistance2  resistance3  resistance4  resistance5  resistance6  spell1  spell2  spell3  spell4  spell5  spell6  spell7  spell8  PetSpellDataId  VehicleId  mingold  maxgold  AIName  MovementType  InhabitType  HoverHeight  HealthModifier  ManaModifier  ArmorModifier  DamageModifier  ExperienceModifier  RacialLeader  movementId  RegenHealth  mechanic_immune_mask  flags_extra  ScriptName       VerifiedBuild
-----  ------------------  ------------------  ------------------  -----------  -----------  --------  --------  --------  --------  -------  -----------  --------  --------------  --------  --------  ------  -------  -------  ----------  ---------  ------  ------  ---------  --------------  ---------------  ------------  -------------  ----------  ----------  -----------  ------------  ------  ------------  -------------  -------------  ------------  ------  ----------  ------  --------------  --------  -----------  -----------  -----------  -----------  -----------  -----------  ------  ------  ------  ------  ------  ------  ------  ------  --------------  ---------  -------  -------  ------  ------------  -----------  -----------  --------------  ------------  -------------  --------------  ------------------  ------------  ----------  -----------  --------------------  -----------  ---------------  -------------
40405                   0                   0                   0            0            0     31834         0         0         0  Kieupid  Pet Trainer                      4783        80        80       0     1604       17           1    1.14286       1       0          0            1500             2000             1              1           2           0         2048             0       0             3              0              0             0       7           0       0               0         0            0            0            0            0            0            0       0       0       0       0       0       0       0       0               0          0        0        0                     0            3            1               1             1              1               1                   1             0           0            1                     0            2  npc_pet_trainer          12340
```
From the `creature` table:
```sql
SELECT * FROM `creature` WHERE `id` = 40405;
```
(no result) - hence no spawned NPC in 3.3.5

**Expected behaviour**: Tell us what should happen instead
The NPC should be spawned in Rangers' Lodge inside Farstriders' Square in Silvermoon City, where the hunter trainers are:
- [Oninath `<Hunter Trainer>`](http://wow.gamepedia.com/Oninath)
- [Tana `<Hunter Trainer>`](http://wow.gamepedia.com/Tana)
- [Zandine `<Hunter Trainer>`](http://wow.gamepedia.com/Zandine)
- [Halthenis `<Pet Trainer>`](http://wow.gamepedia.com/Halthenis)
- [**Kieupid `<Pet Trainer>`**](http://wow.gamepedia.com/Kieupid)

**Steps to reproduce the problem**:

1. Log in with a (horde) character
2. Go to the existing Pet Trainer, Halthenis: `.go creature Halthenis`
3. Notice that Kieupid is nowhere to be found (also check `.go creature Kieupid`)

**Branch(es)**: 3.3.5
(most 6.x pet trainers have been renamed to have their subname `<Pet Trainer>` removed and do not serve any practical purpose than those few who have become quest givers/quest enders)

**TC hash/commit**:  rev. b51409a5d320 2016-08-05 21:22:12 +0200 (3.3.5 branch) (Win64, Release, Static)

**TDB version**:  335.61 plus updates up to and including `2016_08_05_00_world.sql`

**Operating system**:  Windows 10 Home, x64

**Suggested solution:** Add the spawn to the `creature` table:
```sql
-- 40405 Kieupid (Area: 0)
SET @CGUID = XXXXX;

DELETE FROM `creature` WHERE `id` = 40405;
INSERT INTO `creature` (`guid`,`id`,`map`,`zoneId`,`areaId`,`spawnMask`,`phaseMask`,`modelid`,`equipment_id`,`position_x`,`position_y`,`position_z`,`orientation`,`spawntimesecs`,`spawndist`,`currentwaypoint`,`curhealth`,`curmana`,`MovementType`,`npcflag`,`unit_flags`,`dynamicflags`,`VerifiedBuild`) VALUES
(@CGUID, 40405,530,0,0,1,1,0,0, 9924.067, -7400.503, 13.71723, 6.073746, 120,0,0,0,0,0,0,0,0,12340);
```
12547 is the first free GUID as of the time of posting this issue (2016-08-06 01:30 GMT+2).

The query above is tested and works as intended. Full functionality of creature id 40405 will be restored by the SQL content in my PR [**[3.3.5] Core/Scripts: move Pet Trainer gossip to DB**](17746) #17746

[//]: # (This template is for problem reports, for other type of reports edit it accordingly)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing c++ create a Pull Request)

**Comment by Killyana** [comment]
Add the pet: http://www.wowhead.com/npc=40404/tyrantus

**Comment by Aokromes** [comment]
No sniff for it.

**Comment by tkrokli** [comment]
Yes I asked Aokromes for it, but said he does not have any sniff of NPC 40404, only 40405.
(the data above is from 3.3.5 sniff provided by Aokromes)

**Comment by tkrokli** [comment]
`GUID 12547` is still the current lowest available GUID in TDB 335.61 as of this moment.

Suggestion:

```sql
-- NPC entry 40405 Kieupid, Pet Trainer in Silvermoon City (map: 530, zone ID 0, area ID 0)
SET @CGUID = 12547;

DELETE FROM `creature` WHERE `id` = 40405;
INSERT INTO `creature` (`guid`,`id`,`map`,`zoneId`,`areaId`,`spawnMask`,`phaseMask`,`modelid`,`equipment_id`,`position_x`,`position_y`,`position_z`,`orientation`,`spawntimesecs`,`spawndist`,`currentwaypoint`,`curhealth`,`curmana`,`MovementType`,`npcflag`,`unit_flags`,`dynamicflags`,`VerifiedBuild`) VALUES
(@CGUID,40405,530,0,0,1,1,0,0, 9924.067, -7400.503, 13.71723, 6.073746, 120,0,0,0,0,0,0,0,0,12340);
```
